### PR TITLE
Clarify whitespace behavior in read_fwf documentation (#16772)

### LINKED ISSUE #16772: pd.read_fwf removes leading and trailing whitespace
#### Code Sample, a copy-pastable example if possible

```python
from io import StringIO
import pandas as pd

data = u""" a bbb
 ccdd """

df = pd.read_fwf(StringIO(data), widths=[3, 3], header=None)
```

The output is
```python
>>> df.iloc[0,0]
u'a'
```

#### Expected Output
```python
u' a '
```

#### Problem description

Apparently, leading and trailing whitespaces are removed but I want to keep them. Adding dtype options, converters does not solve the problem. Is this expected behaviour?

I do not think this is intended because if we implement the same example with ``pd.read_csv()``, whitespaces are preserved.

```python
from io import StringIO
import pandas as pd

data = u""" a ,bbb
 cc,dd """

df = pd.read_csv(StringIO(data), header=None)
```
```python
>>> df.iloc[0, 0]
' a '
```

For consistency, behaviour should be identical.

The problem is also mentioned on Stackoverflow (https://stackoverflow.com/questions/41558138/pandas-read-fwf-removing-leading-and-trailing-whitespace). 

#### Output of ``pd.show_versions()``

<details>
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.13.final.0
python-bits: 64
OS: Windows
OS-release: 10
machine: AMD64
processor: Intel64 Family 6 Model 69 Stepping 1, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None
LOCALE: None.None

pandas: 0.20.1
pytest: 3.0.7
pip: 9.0.1
setuptools: 27.2.0
Cython: 0.25.2
numpy: 1.12.1
scipy: 0.19.0
xarray: None
IPython: 5.3.0
sphinx: 1.5.6
patsy: 0.4.1
dateutil: 2.6.0
pytz: 2017.2
blosc: None
bottleneck: 1.2.1
tables: 3.2.2
numexpr: 2.6.2
feather: None
matplotlib: 2.0.2
openpyxl: 2.4.7
xlrd: 1.0.0
xlwt: 1.2.0
xlsxwriter: 0.9.6
lxml: 3.7.3
bs4: 4.6.0
html5lib: 0.999
sqlalchemy: 1.1.9
pymysql: None
psycopg2: None
jinja2: 2.9.6
s3fs: None
pandas_gbq: None
pandas_datareader: None

</details>

### PR DESCRIPTION
**Pull request by lphk92**
- [x] closes #16772 
 - [x] tests added / passed
 - [x] passes ``git diff upstream/master -u -- "*.py" | flake8 --diff``
 - [ ] whatsnew entry

### CODE REVIEW

File: pandas/io/parsers.py [hunk 1]
```diff
@@ -316,7 +314,9 @@
     be used automatically. In addition, separators longer than 1 character and
     different from ``'\s+'`` will be interpreted as regular expressions and
     will also force the use of the Python parsing engine. Note that regex
-    delimiters are prone to ignoring quoted data. Regex example: ``'\r\t'``"""
+    delimiters are prone to ignoring quoted data. Regex example: ``'\r\t'``
```
Comment (jreback) [inline pandas/io/parsers.py]: you *may* need to update ``docs/source/io.rst`` as well

File: pandas/tests/io/parser/test_read_fwf.py [hunk 2]
```diff
@@ -405,3 +405,15 @@ def test_skiprows_inference_empty(self):
 
         with pytest.raises(EmptyDataError):
             read_fwf(StringIO(test), skiprows=3)
+
+    def test_whitespace_preservation(self):
+        data_expected = """ a ,bbb
```
Comment (jreback) [inline pandas/tests/io/parser/test_read_fwf.py]: can you add the issue number as a comment

**Comment by jreback** [comment]
cc @chris-b1 

hmm so we are updating the docs? was there a resolution on the issue?

File: pandas/io/parsers.py (hunk 1 above)
Comment (lphk92) [inline pandas/io/parsers.py]: This is done now.

File: doc/source/io.rst [hunk 3]
```diff
@@ -1258,7 +1258,8 @@ Files with Fixed Width Columns
 
 While ``read_csv`` reads delimited data, the :func:`read_fwf` function works
 with data files that have known and fixed column widths. The function parameters
-to ``read_fwf`` are largely the same as `read_csv` with two extra parameters:
+to ``read_fwf`` are largely the same as `read_csv` with two extra parameters, as
+well as a behavioral change in the ``delimiter`` parameter:
```
Comment (jreback) [inline doc/source/io.rst]: what does this addition mean?

Comment (lphk92) [inline doc/source/io.rst]: That the delimiter parameter behaves differently in read_fwf than it does in read_csv (as specified in the new documentation)

File: pandas/io/parsers.py [hunk 4]
```diff
@@ -316,7 +314,9 @@
...
+delimiter : str, default ``None``
+    Alternative argument name for sep."""
```
Comment (gfyoung) [inline pandas/io/parsers.py]: Why was this moved in the first place?

File: doc/source/io.rst (hunk 3 above)
Comment (gfyoung) [inline doc/source/io.rst]: I would reword as : "and a different usage of the ``delimiter`` parameter" instead of "as well as...parameter"

**Comment by gfyoung** [comment]
@jreback : AFAICT, I think @chris-b1 was thinking we should better surface and document the `delimiter` parameter, which I agree with and is what this PR does.

File: doc/source/io.rst (hunk 3 above)
Comment (lphk92) [inline doc/source/io.rst]: Done.

File: pandas/tests/io/parser/test_read_fwf.py [hunk 5]
```diff
@@ -405,3 +405,16 @@ def test_skiprows_inference_empty(self):
...
         with pytest.raises(EmptyDataError):
             read_fwf(StringIO(test), skiprows=3)
+
+    def test_whitespace_preservation(self):
+        # Addresses Issue #16772
+        data_expected = """ a ,bbb
+ cc,dd """
+        expected = read_csv(StringIO(data_expected), header=None)
+
+        test_data = """ a bbb
```
Comment (jreback) [inline pandas/tests/io/parser/test_read_fwf.py]: can you also add an example using ``\t`` (the default)

File: pandas/tests/io/parser/test_read_fwf.py [hunk 6]
```diff
@@ -405,3 +405,16 @@ def test_skiprows_inference_empty(self):
...
+ ccdd """
+        df = read_fwf(StringIO(test_data), widths=[3, 3],
```
Comment (jreback) [inline pandas/tests/io/parser/test_read_fwf.py]: result =

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/io/parsers.py [hunk 7]
```diff
@@ -341,15 +341,16 @@
 widths : list of ints. optional
     A list of field widths which can be used instead of 'colspecs' if
     the intervals are contiguous.
+delimiter: str, default ``\t ``
```
Comment (chris-b1) [inline pandas/io/parsers.py]: Wrap default value in quotes - at a glance it's kind of trick to see that whitespace is included maybe write as `'\t' + ' '`

**Comment by chris-b1** [comment]
This looks good to me, except for doc notes.  I didn't know the delimiter kw worked this way on `read_fwf`, so not what I had in mind, but is the right solution!

File: pandas/tests/io/parser/test_read_fwf.py (hunk 5 above)
Comment (lphk92) [inline pandas/tests/io/parser/test_read_fwf.py]: Done

File: pandas/io/parsers.py (hunk 7 above)
Comment (lphk92) [inline pandas/io/parsers.py]: Done

File: pandas/tests/io/parser/test_read_fwf.py (hunk 6 above)
Comment (lphk92) [inline pandas/tests/io/parser/test_read_fwf.py]: Not entirely sure what you're asking for. Do you just want me to rename 'df' to 'result'?

Comment (gfyoung) [inline pandas/tests/io/parser/test_read_fwf.py]: Yes, that is correct.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
lgtm. merge on green.

**Comment by gfyoung** [comment]
@lphk92 : If you could address my question <a href="https://github.com/pandas-dev/pandas/pull/16950#discussion_r127601576">above</a> before merging this, that would be great!

File: pandas/io/parsers.py (hunk 4 above)
Comment (lphk92) [inline pandas/io/parsers.py]: I'm not sure why this was originally moved out of `_sep_doc`, however I've put it back as this particular documentation of the `delimiter` parameter only applies to `read_csv` and `read_table`, both of which use `_sep_doc` in their documentation.

**Comment by lphk92** [comment]
@gfyoung I've just responded to your question as best I can. Let me know if I can provide more clarification.

File: pandas/io/parsers.py (hunk 4 above)
Comment (gfyoung) [inline pandas/io/parsers.py]: Awesome.  Thanks!

**Comment by gfyoung** [comment]
Alright, everything looks good now.  Merging.

**Comment by gfyoung** [comment]
Thanks @lphk92 !
